src/CMakeLists.txt

```diff
@@ -429,10 +435,6 @@ if(HAVE_RDMACM)
   list(APPEND ceph_common_deps RDMA::RDMAcm)
 endif()
 
-if(NOT WITH_SYSTEM_BOOST)
```

Comment:
Nit: is there a reason this moved up to be unconditionally included?

Reply:
@petrutlucian94 i think a better fix is #36696. could you help test it?

Reply:
Sure, I'll give it a try.

Reply:
Yep, works. Still, I'm wondering if this should also be done when linking dynamically.

Reply:
no, i don't think we need to worry about the dynamically linked boost. as the linker does not need to include the dynamically linked library when linking it, i think. actually, that's what we are doing when building Ceph on ubuntu/bionic builders where we link against prebuilt boost shared libraries.

Reply:
DLLs are a bit more strict, requiring all the symbols to be located at build time. Anyway, we're only using static builds for now when targeting Windows, mostly due to certain objects exceeding the maximum allowed number of symbols. We can revisit it later if needed.

I'm updating the PR, dropping this specific change and rely on https://github.com/ceph/ceph/pull/36696 instead.